tests/ec2.rs

```diff
@@ -23,3 +24,47 @@ fn main() {
     }
 
 }
+
+#[test]
+// Test for issue 383
+fn dry_run() {
+    let credentials = DefaultCredentialsProvider::new().unwrap();
+    let ec2 = Ec2Client::new(credentials, Region::UsEast1);
+    let req = CreateSnapshotRequest {
+                     volume_id: "v-00000001".into(),
+                     description: None,
+                     dry_run: Some(true)
+                 };
+    match ec2.create_snapshot(&req) {
+        Ok(_) => {
+            panic!("create_snapshot should fail");
```

Comment:
Would it be possible for the test to be re-written to use [`should_panic`](https://doc.rust-lang.org/book/testing.html)? It's what's typically used for tests of this format, but I'm not sure if `should_panic`'s `expected=""` can match against error descriptions themselves.

I'm thinking of something like this:

```rust
#[test]
#[should_panic(expected = "<Message>Request would have succeeded, but DryRun flag is set.</Message>")]
fn dry_run() {
    ...

    ec2.create_snapshot(&req).unwrap();
}
```

(Ditto for the test below)